CRM/Contribute/Form/Contribution.php

```diff
@@ -998,7 +998,12 @@ public static function formRule($fields, $files, $self) {
         $errors['trxn_id'] = ts('Transaction ID\'s must be unique. Transaction \'%1\' already exists in your database.', array(1 => $fields['trxn_id']));
       }
     }
-
+    // CRM-16189
+    $errorMessage = CRM_Financial_BAO_FinancialAccount::checkFinancialTypeHasDeferred($fields, $self->_id, $self);
```

Comment:
IMO `checkFinancialTypeHasDeferred` should be updated to return a boolean and the message should be generated here.